bug: title-editor.tsx 触发 react-hooks/set-state-in-effect 致 npm run lint 失败

## Repro

```bash
git checkout main
npx eslint 'app/(app)/cases/[id]/title-editor.tsx'
# or:
npm run lint
```

## Result

```
app/(app)/cases/[id]/title-editor.tsx
  22:5  error  Calling setState synchronously within an effect can trigger cascading renders
                react-hooks/set-state-in-effect
✖ 1 problem (1 error, 0 warnings)
```

## Code involved

`title-editor.tsx`（行 21-23）使用 `useEffect` 同步 `initialTitle` prop 到本地 `draft` state：

```tsx
useEffect(() => {
  setDraft(initialTitle);
}, [initialTitle]);
```

## Why it matters

- `npm run lint` 在 main 上即非零退出，会阻塞依赖 lint pass 的 CI / commit gate
- 已影响 #21 的 agent brief 验收（agent 需要在不修改无关代码的前提下，无法满足 "lint 通过" 这条 acceptance criterion）

## Source

由 `60ef08f feat(cases): AI auto-generate title with inline editing + regenerate proposal` 引入；`dbcf4a0` 之后未再修改。

## Suggested fix direction

React 文档（[You Might Not Need an Effect](https://react.dev/learn/you-might-not-need-an-effect)）正是针对此模式：把 `draft` 改成由 `initialTitle` 派生的 state（用 `key` 强制重置子组件，或在渲染期发现 prop 变化时直接 `setDraft`）。具体取舍由实施者判断。

--- a/app/(app)/cases/[id]/title-editor.tsx
+++ b/app/(app)/cases/[id]/title-editor.tsx
@@ -1,7 +1,7 @@
 "use client";
 
 import { useRouter } from "next/navigation";
-import { useEffect, useState, useTransition } from "react";
+import { useState, useTransition } from "react";
 import { Check, Loader2, Pencil, RefreshCcw, X } from "lucide-react";
 import { updateCaseTitle } from "@/app/(app)/cases/actions";
 import { Input } from "@/components/ui/input";
@@ -15,12 +15,14 @@ export function TitleEditor({ proposalCaseId, initialTitle }: TitleEditorProps) 
   const router = useRouter();
   const [isPending, startTransition] = useTransition();
   const [editing, setEditing] = useState(false);
+  const [previousInitialTitle, setPreviousInitialTitle] = useState(initialTitle);
   const [draft, setDraft] = useState(initialTitle);
   const [regenerating, setRegenerating] = useState(false);
 
-  useEffect(() => {
+  if (initialTitle !== previousInitialTitle) {
+    setPreviousInitialTitle(initialTitle);
     setDraft(initialTitle);
-  }, [initialTitle]);
+  }
 
   const handleSave = async () => {
     if (!draft.trim() || draft === initialTitle) {

